Code/Framework/AzToolsFramework/Tests/ComponentModeSwitcherTests.cpp

```diff
@@ -0,0 +1,423 @@
+        EXPECT_TRUE(componentModeSwitcher->GetComponentCount() == 2);
+
+        // Then if one component is disabled, there should only be one component on the switcher
+        AzToolsFramework::EntityCompositionRequestBus::Broadcast(
+            &AzToolsFramework::EntityCompositionRequests::DisableComponents, AZStd::vector<AZ::Component*>{ placeholder1 });
+
+        EXPECT_TRUE(componentModeSwitcher->GetComponentCount() == 1);
+    }
+
+    TEST_F(ComponentModeSwitcherTestFixture, EnablingComponentAddsComponentToSwitcher)
+    {
+        // Given an entity with two components
+        AZStd::shared_ptr<ComponentModeSwitcher> componentModeSwitcher = AZStd::make_shared<ComponentModeSwitcher>();
+
+        AzToolsFramework::EditorTransformComponentSelectionRequestBus::Event(
+            AzToolsFramework::GetEntityContextId(),
+            &AzToolsFramework::EditorTransformComponentSelectionRequestBus::Events::OverrideComponentModeSwitcher,
+            componentModeSwitcher);
+        AZ::Entity* entity = nullptr;
+        AZ::EntityId entityId = CreateDefaultEditorEntity("ComponentModeEntity", &entity);
+
+        entity->Deactivate();
+        const AZ::Component* placeholder1 = entity->CreateComponent<PlaceholderEditorComponent>();
+        [[maybe_unused]] const AZ::Component* placeholder2 = entity->CreateComponent<AnotherPlaceholderEditorComponent>();
+        entity->Activate();
+
+        // When the component is disabled it doesn't show up in the switcher
+        const AzToolsFramework::EntityIdList entityIds = { entityId };
+        AzToolsFramework::ToolsApplicationRequestBus::Broadcast(
+            &AzToolsFramework::ToolsApplicationRequests::SetSelectedEntities, entityIds);
+        EXPECT_TRUE(componentModeSwitcher->GetComponentCount() == 2);
+
+        AzToolsFramework::EntityCompositionNotificationBus::Broadcast(
+            &AzToolsFramework::EntityCompositionNotificationBus::Events::OnEntityComponentDisabled, entity->GetId(), placeholder1->GetId());
+        EXPECT_TRUE(componentModeSwitcher->GetComponentCount() == 1);
+
+        // Then if the component is enabled again, both components show in the switcher
+        //AzToolsFramework::EntityCompositionRequestBus::Broadcast(
+        //    &AzToolsFramework::EntityCompositionRequests::EnableComponents, AZStd::vector<AZ::Component*>{ placeholder1 });
+        AzToolsFramework::EntityCompositionNotificationBus::Broadcast(
+            &AzToolsFramework::EntityCompositionNotificationBus::Events::OnEntityComponentEnabled, entity->GetId(), placeholder1->GetId());
+
+        AzToolsFramework::ToolsApplicationRequestBus::Broadcast(
+            &AzToolsFramework::ToolsApplicationRequests::SetSelectedEntities, AzToolsFramework::EntityIdList{});
+        AzToolsFramework::ToolsApplicationRequestBus::Broadcast(
+            &AzToolsFramework::ToolsApplicationRequests::SetSelectedEntities, entityIds);
```

Comment:
Why are there two consecutive calls to SetSelectedEntities?
The latter call should be sufficient.